src/domain/mongoose_subdomain_utils.erl

```diff
@@ -0,0 +1,32 @@
+-module(mongoose_subdomain_utils).
+
+%% API
+-export([make_subdomain_pattern/1,
+         get_fqdn/2,
+         is_subdomain/3]).
+
+-type subdomain_pattern() :: {fqdn | prefix, binary()}.
+-export_type([subdomain_pattern/0]).
+
+-spec make_subdomain_pattern(SubdomainPatternConfigOpt :: binary() | string()) ->
```

Comment:
maybe some comment on what it does and why is needed.

Reply:
added